### ISSUE TITLE
[DLB]2018-02-03 질문사항

### ISSUE THREAD

**Issue by SongChiYoung** [YBIGTA/data-science-2018#11]
https://github.com/YBIGTA/DLB/blob/master/Neural%20Networks%20and%20Deep%20Learning/3.%20Shallow%20Neural%20Network.md

**Comment by KangBK0120** [comment]
궁금한 점에 대해서는 [이 글](https://www.quora.com/What-is-the-dying-ReLU-problem-in-neural-networks)을 읽어보시면 좋을 것 같습니다. 말씀하신 문제는 Dying ReLU라고 볼 수 있는데요. 실제로 얼마나 이런 경우가 생기는지는 모르겠지만

1. Learning Rate를 조절하든가
2. Leaky ReLU를 사용한다든가

하는 방법으로 해결한답니다. 이미 죽어버린 ReLU Cell도 학습이 진행되면서 다시 살아날 수 있기는 합니다

**Comment by hyejukim-ybigta** [comment]
Sigmoid function은 RL에서 output layer에서 사용하고, 그 이외에는 잘 사용하지 않습니다. >>라고 해주셨는데 RL에서도 ReLU function 과 같은 것을 이용하면 성능이 더 좋아지나요?

**Comment by statkim7578** [comment]
2 layer RL Neural Network의 Gradient Descent(GD) 부분이 통째로 잘 이해가 안되네요 ㅠㅠ
발제때 이 부분 천천히 설명해주시면 잘 따라가겠습니다

**Comment by ybigtaJH** [comment]
정규분포에 따라서 추출한다고 하셨는데, 숫자의 범위가 어떻게 되나요?

**Comment by DabinJeong** [comment]
만약 tanh를 activation function으로 쓰는 거라면 다음 층의 노드들에는 -1~1사이의 값이 들어가는 건가요??
그리고 activation function은 모든 노드들 끝에 있는 건가요??

**Comment by godongyoung** [comment]
' z가 0일때 미분불가능하지만 보통 0이라고 합니다.(구글링) 1이라고 해도 상관이 없는데 그 이유는 z값이 작으면 어차피 gradient가 매우 작이지므로 ( dw = xdz에서 x의 역할) 크게 영향을 받지 않기 때문이라고 합니다. '
이 부분이 잘 이해가 되지 않습니다 미분값을 0으로 해도되고 1로해도 된다는건가요??
이해햇습니다 하하 추가로 z=(정확히)0이 나올 확률은 0에 수렴해서 큰 관심이 아니다라고합니당

**Comment by Jooong** [comment]
@hyejukim-ybigta 기본적으로 sigmoid는 [saturating non-linearity](https://stats.stackexchange.com/questions/174295/what-does-the-term-saturating-nonlinearities-mean)기 때문에 역전파 과정에서 gradient가 잘 전달되지 않습니다(Vanishing gradient.) 이런 관점에서 ReLU가 월등히 성능이 좋기 때문에, 보편적으로 ReLU를 activation으로 사용합니다. sigmoid같은 경우 언급된 것과 같이 맨 마지막에 값을 0~1의 확률로 매핑하기 위한 용도로 주로 사용되는 것 같습니다.
